WordPress/src/main/java/org/wordpress/android/ui/reader/subfilter/SubFilterSharedViewModel.kt

```diff
@@ -0,0 +1,225 @@
+import org.wordpress.android.modules.UI_THREAD
+import org.wordpress.android.ui.Organization
+import org.wordpress.android.ui.Organization.NO_ORGANIZATION
+import org.wordpress.android.ui.prefs.AppPrefsWrapper
+import org.wordpress.android.ui.reader.ReaderEvents
+import org.wordpress.android.ui.reader.services.update.ReaderUpdateLogic.UpdateTask
+import org.wordpress.android.ui.reader.subfilter.SubfilterListItem.Site
+import org.wordpress.android.ui.reader.subfilter.SubfilterListItem.SiteAll
+import org.wordpress.android.ui.reader.subfilter.SubfilterListItem.Tag
+import org.wordpress.android.ui.reader.subfilter.SubfilterVisibilityUiModel.SubfilterHide
+import org.wordpress.android.ui.reader.subfilter.SubfilterVisibilityUiModel.SubfilterShow
+import org.wordpress.android.util.AppLog
+import org.wordpress.android.util.AppLog.T
+import org.wordpress.android.util.EventBusWrapper
+import org.wordpress.android.viewmodel.Event
+import org.wordpress.android.viewmodel.ScopedViewModel
+import java.util.EnumSet
+import javax.inject.Inject
+import javax.inject.Named
+
+class SubFilterSharedViewModel @Inject constructor(
+    @Named(UI_THREAD) private val mainDispatcher: CoroutineDispatcher,
+    @Named(BG_THREAD) private val bgDispatcher: CoroutineDispatcher,
+    private val appPrefsWrapper: AppPrefsWrapper,
+    private val eventBusWrapper: EventBusWrapper,
+    private val accountStore: AccountStore
+) : ScopedViewModel(bgDispatcher) {
+    private val _subFilters = MutableLiveData<List<SubfilterListItem>>()
+    val subFilters: LiveData<List<SubfilterListItem>> = _subFilters
+
+    private val _changeBottomSheetVisibility = MutableLiveData<Event<SubfilterVisibilityUiModel>>()
+    val changeBottomSheetVisibility: LiveData<Event<SubfilterVisibilityUiModel>> = _changeBottomSheetVisibility
+
+    private val _updateTagsAndSites = MutableLiveData<Event<EnumSet<UpdateTask>>>()
+    val updateTagsAndSites: LiveData<Event<EnumSet<UpdateTask>>> = _updateTagsAndSites
+
+    private val _onSubfilterClicked = MutableLiveData<HashMap<Organization, Event<SubfilterListItem>>>()
+    val onSubfilterClicked: LiveData<HashMap<Organization, Event<SubfilterListItem>>> = _onSubfilterClicked
+
+    private val _filtersMatchCount = MutableLiveData<HashMap<SubfilterCategory, Int>>()
+    val filtersMatchCount: LiveData<HashMap<SubfilterCategory, Int>> = _filtersMatchCount
+
+    private val _bottomSheetEmptyViewAction = MutableLiveData<Event<ActionType>>()
+    val bottomSheetEmptyViewAction: LiveData<Event<ActionType>> = _bottomSheetEmptyViewAction
+
+    private var isStarted = false
+
+    private var lastKnownUserId: Long? = null
+    private var lastTokenAvailableStatus: Boolean? = null
+
+    fun start() {
+        if (isStarted) {
+            return
+        }
+
+        isStarted = true
+
+        _filtersMatchCount.value = hashMapOf()
+        _onSubfilterClicked.value = hashMapOf()
+
+        eventBusWrapper.register(this)
+    }
+
+    fun loadSubFilters() {
```

Comment:
Is the only reason we are using a shared VM that we don't want to invoke this method multiple times when "FollowedTagsChanged" or "FollowedBlogsChanged" is emitted. 

If it's not the case, ignore the rest of this comment:). If it is the case I think that it might be an over-optimization. Loading data from the local database is quite cheap and this method performs it on a background thread. So it shouldn't affect the user at all.